src/Our.Umbraco.Ditto/Extensions/TypeInferenceExtensions.cs

```diff
@@ -90,5 +92,33 @@ public static IEnumerable<Type> GetBaseTypes(this Type type)
                 type = type.BaseType;
             }
         }
+
+        /// <summary>
+        /// Gets a all Type instances matching the specified class name with just non-namespace qualified class name.
+        /// </summary>
+        /// <param name="className">Name of the class sought.</param>
+        /// <param name="baseType"></param>
+        /// <returns>Types that have the class name specified. They may not be in the same namespace.</returns>
+        public static IEnumerable<Type> GetTypeByName(string className, Type baseType)
+        {
+            /* [ML] - Look away kids, forgive me. We could improve performance with an optional app setting to filter assemblies 
+             * but this method executes in low milliseconds, hopefully with caching of these results we'll be ok
+             */
+
+            var assemblies = AppDomain.CurrentDomain.GetAssemblies().Where(i => !i.FullName.ToLower().StartsWith("system") && !i.FullName.ToLower().StartsWith("microsoft") && !i.FullName.ToLower().StartsWith("windows")).ToList();
```

Comment:
Ah... Umbraco already has a method that can provide inherited types. I would imagine (I haven't checked the source) that it's highly optimised.

PluginManager.Current.ResolveTypes<YOUR_BASE_TYPE>()